src/libexpr/value.hh

```diff
@@ -13,6 +13,21 @@
 
 #include <nlohmann/json_fwd.hpp>
 
+#include <immer/heap/gc_heap.hpp>
+#include <immer/heap/heap_policy.hpp>
+#include <immer/memory_policy.hpp>
+#include <immer/refcount/no_refcount_policy.hpp>
+#include <immer/flex_vector.hpp>
+#include <immer/flex_vector_transient.hpp>
+
+// declare a memory policy for using a tracing garbage collector
+using gc_policy = immer::memory_policy<immer::heap_policy<immer::gc_heap>,
+                                       immer::no_refcount_policy,
+                                       immer::default_lock_policy,
```

Comment:
I wonder what the default lock policy does.
I suppose clever locking might save allocations in a few cases, but I'm inclined to think that just allocating all the time is more efficient, because we tend to keep the "old version" around because purely functional operations are non-destructive and our scopes/closures (`Env`) tend to stick around fairly often in practice, and if they don't, we don't really know locally.

Reply:
The locking is not used for allocation actually, but it's used by `immer::atom` so no need to worry about it :)